File: python/1711_CountGoodMeals.py
class Solution:
    def countPairs(self, deliciousness: List[int]) -> int:
        high = 1000000007
        counts = defaultdict(int)
        pairs = 0
        for n in deliciousness:
            for i in range(22):
                pairs = (pairs + counts[(1 << i) - n]) % high
            counts[n] += 1
        return pairs

File: swift/1711_CountGoodMeals.swift
class Solution {
    func countPairs(_ deliciousness: [Int]) -> Int {
        var high = 1000000007
        var deliciousnessCounts = [Int:Int]()
        var counts = 0
        for n in deliciousness {
            for i in 0..<22 {
                counts = (counts + (deliciousnessCounts[(1 << i) - n] ?? 0)) % high
            }
            deliciousnessCounts[n] = (deliciousnessCounts[n] ?? 0) + 1
        }
        return counts
    }
}
